evap/staff/importers.py

```diff
@@ -120,6 +132,55 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
             textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
         )
 
+    def check_for_existing_course(self, semester):
+        course = Course.objects.filter(
+            semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
+        ).first()
+
+        if course:
+            merge_hindrances = self.course_merge_hindrances(course)
+            if merge_hindrances:
+                return merge_hindrances
+            self.existing_course = course
+            return set()
+
+        name_collision = self.check_name_collision(semester)
+        if name_collision == NameCollision.NONE:
+            return set()
+        if name_collision == NameCollision.NAME_DE:
+            return set([_("the german course name does not match")])
+        if name_collision == NameCollision.NAME_EN:
+            return set([_("the english course name does not match")])
```

Comment:
With this, the final message to the user would be:
> Course Shake (Schütteln) does already exist in this semester, but the courses can not be merged for the following reasons:
> - the german course name does not match

and, similarly,
> Course Shake (Schütteln) does already exist in this semester, but the courses can not be merged for the following reasons:
> - the english course name does not match

For the user, this requires quite some thinking to figure out what is actually going on, compared to
> Course Shake (english name) already exists in this semester with differing german name

and
>Course Schütteln (german name) already exists in this semester with differing english name

I'd argue: Move this into the importer, so we have access to self.errors here and are not forced to have one unified message constructed around it.

Then, I'd think we can more easily find a name, such as `set_existing_course_to_merge_or_add_errors`, or simply `handle_course_name_collisions`, or whatever floats your boat.

Reply:
(I'd let you and @niklasmohrin overrule me here if you disagree, because I'm afraid I might be unfair here. Background: I've played around with this for a while for https://github.com/e-valuation/EvaP/pull/1683#issuecomment-1039503929 and think the general control flow structure there is close to the best we could achieve for simplest code with best user messages, apart from whether it is in an extra method or directly in the importer's loop. But, since it's my code, I'm probably biased, thus the overruling.)

Reply:
Yep, understand your point and agree, see https://github.com/e-valuation/EvaP/pull/1683/commits/79d7b1f845d6f8beab446440f2cbaaf0b8ac9ae6